scripts/tslint/preferConstRule.ts

```diff
@@ -106,18 +106,23 @@ class PreferConstWalker extends Lint.RuleWalker {
         if (node.kind === ts.SyntaxKind.ObjectLiteralExpression) {
             const pattern = node as ts.ObjectLiteralExpression;
             for (const element of pattern.properties) {
-                if (element.name.kind === ts.SyntaxKind.Identifier) {
-                    this.markAssignment(element.name as ts.Identifier);
+                const kind = element.kind;
+
+                if (kind === ts.SyntaxKind.ShorthandPropertyAssignment) {
+                    this.markAssignment((element as ts.ShorthandPropertyAssignment).name);
+                }
+                else if (kind === ts.SyntaxKind.PropertyAssignment) {
+                    this.visitLeftHandSideExpression((element as ts.PropertyAssignment).initializer);
                 }
-                else if (isBindingPattern(element.name)) {
-                    this.visitBindingPatternIdentifiers(element.name as ts.BindingPattern);
+                else {
+                    // Should we throw an exception?

```

Comment:
The only thing that we can parse as property of `ObjectLiteralExpression` except `PropertyAssignment` and `ShorthandPropertyAssignment` is `MethodDeclaration` and it is already a grammar error to have it in destructuring assignment. So I think it is safe to ignore other syntax kinds